knative-operator/pkg/controller/knativeserving/consoleclidownload/consoleclidownload.go

```diff
@@ -154,75 +153,52 @@ func Delete(instance *servingv1alpha1.KnativeServing, apiclient client.Client, s
 		return fmt.Errorf("failed to delete kn ConsoleCLIDownload CO: %w", err)
 	}
 
-	log.Info("Deleting kn ConsoleCLIDownload resources")
-	manifest, err := manifest(instance, apiclient, scheme)
-	if err != nil {
-		return err
-	}
-
-	if err := manifest.Delete(); err != nil && !apierrors.IsNotFound(err) {
-		return fmt.Errorf("failed to delete kn ConsoleCLIDownload resources manifest: %w", err)
+	log.Info("Deleting kn ConsoleCLIDownload Service")
+	if err := apiclient.Delete(context.TODO(), makeKnService("", instance)); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to delete kn ConsoleCLIDownload Service: %w", err)
```

Comment:
If you go by my suggestion below, this deletion will be handled via the OwnerReference and we don't need to "manually" delete this service here.

Reply:
Keeping the "manual" delete due to reason above.